dcos/package.py

```diff
@@ -943,6 +944,25 @@ def copy_to_cache(self, target_dir):
                 'Unable to fetch packages from [{}]'.format(self.url))
 
 
+def _onerror(func, path, exc_info):
+    """
+    Error handler for ``shutil.rmtree``.
+
+    If the error is due to an access error (read only file)
+    it attempts to add write permission and then retries.
+
+    If the error is for another reason it re-raises the error.
+
+    Usage : ``shutil.rmtree(path, onerror=onerror)``
+    """
+    import stat
+    if not os.access(path, os.W_OK):
+        os.chmod(path, stat.S_IRWXU | stat.S_IRWXG | stat.S_IRWXO)
+        func(path)
+    else:
+        raise
```

Comment:
`raise` without arguments - is a common way for re-raising exceptions. One more reason, is changes in py3 for exception raising. If raise exception manually using exc_info, then such `During handling of the above exception, another exception occurred` message will be shown. Py3 identifying this exception like new, and not like re-raising old one.